packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -554,93 +374,84 @@ export const useUpdateCounterpartContact = (counterpartId: string) => {
   });
 };
 
-export const useDeleteCounterpartContact = (counterpartId: string) => {
+export const useDeleteCounterpartContact = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { invalidate } = useCounterpartListCache();
-  const { remove } = useCounterpartContactListCache(counterpartId);
-
-  return useMutation<void, Error, string>({
-    mutationFn: (contactId) =>
-      monite.api.counterparts.deleteContact(counterpartId, contactId),
-
-    onSuccess: (_, contactId) => {
-      remove(contactId);
-      invalidate();
-
-      toast.success(t(i18n)`Contact Person was deleted.`);
-    },
-
-    onError: (error) => {
-      toast.error(
-        getLegacyAPIErrorMessage(error) ||
-          t(i18n)`Failed to delete Contact Person.`
-      );
-    },
-  });
+  const { api } = useMoniteContext();
+
+  return api.counterparts.deleteCounterpartsIdContactsId.useMutation(
+    undefined,
+    {
+      onSuccess: () => {
+        toast.success(t(i18n)`Contact Person was deleted.`);
+      },
+
+      onError: (error) => {
+        toast.error(
+          getLegacyAPIErrorMessage(error) ||
+            t(i18n)`Failed to delete Contact Person.`
+        );
+      },
+    }
+  );
 };
 
 export const useCounterpartList = (
-  ...args: Parameters<CounterpartsService['getList']>
+  parameters?: Services['counterparts']['getCounterparts']['types']['parameters']
 ) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartPaginationResponse, Error>({
-    queryKey: [...counterpartQueryKeys.list(), ...args] as const,
-    queryFn: () => monite.api.counterparts.getList(...args),
-  });
+  const { api } = useMoniteContext();
+  return api.counterparts.getCounterparts.useQuery(parameters ?? {});
 };
 
 export const useCreateCounterpart = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { invalidate } = useCounterpartListCache();
-  const { setEntity } = useCounterpartDetailCache();
-
-  return useMutation<CounterpartResponse, Error, CounterpartCreatePayload>({
-    mutationFn: (payload) => monite.api.counterparts.create(payload),
-
-    onSuccess: (counterpart) => {
-      setEntity(counterpart);
-      invalidate();
-
-      toast.success(
-        t(i18n)`Counterpart “${getCounterpartName(counterpart)}” was created.`
-      );
-    },
+  const { api } = useMoniteContext();
+  const queryClient = useQueryClient();
+
+  return api.counterparts.postCounterparts.useMutation(
+    {},
+    {
+      onSuccess: async (counterpart) => {
+        await api.counterparts.getCounterpartsId.invalidateQueries(queryClient);
```

Comment:
`getCounterpartsId` invalidates all `/counterparts/{counterpart_id}` endpoints, which is not the case. You need to invalidate all `GET /counterparts`:

```ts
await api.counterparts.getCounterparts.invalidateQueries(queryClient);
```